frontend/server/src/Controllers/Problem.php

```diff
@@ -4529,6 +4529,17 @@ public static function getProblemDetailsForTypeScript(
                 ),
             ]
         );
+
+        $nextSubmissionTimestamp = \OmegaUp\DAO\Runs::nextSubmissionTimestampByProblem(
+            intval($problem->problem_id),
+            intval($r->identity->identity_id)
+        );
+
+        $response['smartyProperties']['payload']['problem'] = array_merge(
+            $response['smartyProperties']['payload']['problem'],
+            ['nextSubmissionTimestamp' => $nextSubmissionTimestamp]
+        );
```

Comment:
```suggestion
        $response['smartyProperties']['payload']['problem']['nextSubmissionTimestamp']  = \OmegaUp\DAO\Runs::nextSubmissionTimestampByProblem(
            intval($problem->problem_id),
            intval($r->identity->identity_id)
        );
```

también, aquí sí se vale hacer lo de mover obtener todos los envíos acá y obtener el último en PHP. así nos ahorramos un viaje redondo a la base de datos.